std/uni.d

```diff
@@ -8843,16 +8850,6 @@ void toLowerInPlace(C)(ref C[] s) @trusted pure
 {
     toCaseInPlace!(LowerTriple)(s);
 }
-// overloads for the most common cases to reduce compile time
-@safe pure /*TODO nothrow*/
```

Comment:
When I first read your comment, I didn't realize that you were removing all of those overloads. I thought that you were just removing the attributes and that the main overload was more generic. However, it looks like the main overload is specifically for strings and that the point of the other overloads is to somehow improve compilation time by making the primary set of possible argument types explicit. My guess is that it results in `toLower` being instantiated only once for each of the main string types, whereas without those overloads, it would be instantiated each time it was used and that that results in faster compilation, but I don't know.

I agree that it's ugly that they're there, but I would assume that Martin wouldn't have added them if there weren't a significant gain, and I definitely don't think that they should be removed without his input.

As for the attributes, the main overload already has them. I made the mistake of assuming that what showed above the comment was the full section of code that was being removed and was looking at my local copy of the file rather than the full diff, so I made an ill-informed suggestion.